test/unittests/parser_test.cpp

```diff
@@ -595,27 +596,29 @@ TEST(parser, milestone1)
 TEST(parser, instr_loop)
 {
     const auto loop_without_imm = "030b"_bytes;
-    EXPECT_THROW(parse_expr(loop_without_imm.data()), parser_error);
-
-    const auto loop_missing_end = "03400b"_bytes;
-    EXPECT_THROW(parse_expr(loop_missing_end.data()), parser_error);
+    EXPECT_THROW_MESSAGE(
+        parse_expr(loop_without_imm.data()), parser_error, "loop can only have type arity 0");
 
     const auto loop_void_empty = "03400b0b"_bytes;
     const auto [code1, pos1] = parse_expr(loop_void_empty.data());
     EXPECT_EQ(code1.instructions, (std::vector{Instr::loop, Instr::end, Instr::end}));
     EXPECT_EQ(code1.immediates.size(), 0);
 
     const auto loop_i32_empty = "037f0b0b"_bytes;
-    EXPECT_THROW(parse_expr(loop_i32_empty.data()), parser_error);  // loop arity != 0.
+    EXPECT_THROW_MESSAGE(
+        parse_expr(loop_i32_empty.data()), parser_error, "loop can only have type arity 0");
+}
+
+TEST(parser, DISABLED_instr_loop_input_buffer_overflow)
```

Comment:
Why is this causing a buffer overflow?

Reply:
Because the end opcode is missing, and it continues until finds random byte which is a valid opcode.

Reply:
As a reminder, `03400b` is consumed as `loop <0x40> end` and should fail with "loop can only have type arity 0". Why isn't that happening?

Reply:
Ah it's missing the real `end` opcode.

Reply:
Can you leave that as a comment?

Reply:
What exactly?

Reply:
Don't know, tried to describe but its long: "The `loop` instruction is well formed as it contains the closing `end`, but the code block doesn't contain the implicit closing `end`."

Perhaps just add "`03400b0b`" would be valid.